cli/Dockerfile

```diff
@@ -1,24 +1,40 @@
-# Kilo Code CLI Docker Image
-FROM node:20.19.2-bookworm-slim
-
 # Build arguments for metadata (all optional with defaults)
-ARG BUILD_DATE=""
-ARG VCS_REF=""
-ARG VERSION="latest"
-
-# OCI Image labels
-LABEL org.opencontainers.image.created="${BUILD_DATE}"
-LABEL org.opencontainers.image.revision="${VCS_REF}"
-LABEL org.opencontainers.image.version="${VERSION}"
-LABEL org.opencontainers.image.title="Kilo Code CLI"
-LABEL org.opencontainers.image.description="Docker image for Kilo Code CLI with Chromium support"
-LABEL org.opencontainers.image.vendor="Kilo Code"
-LABEL org.opencontainers.image.url="https://kilocode.ai/docs/cli"
-LABEL org.opencontainers.image.documentation="https://kilocode.ai/docs/cli"
-LABEL org.opencontainers.image.source="https://github.com/Kilo-Org/kilocode"
-LABEL org.opencontainers.image.licenses="Apache-2.0"
-
-# Install system dependencies and clean up in one layer
+ARG BUILD_DATE
+ARG VCS_REF
+ARG VERSION
+
+# ==========================================
+# 1. Alpine Base 
+# ==========================================
+FROM node:20.19.2-alpine AS alpine_base
+
+# Install dependencies
+RUN apk add --no-cache \
+    # Build tools for native npm modules
+    python3 \
+    make \
+    g++ \
+    # Version control
+    git \
+    # Network tools
+    curl
+
+# Create non-root user and set up directories
+RUN addgroup -S kilocode && \
+    adduser -S -G kilocode -G audio -G video kilocode && \
+    mkdir -p /home/kilocode/.kilocode /workspace && \
+    chown -R kilocode:kilocode /home/kilocode /workspace
+
+# Set environment variables
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true \
+    HOME=/home/kilocode
```

Comment:
For the deb we're installing chromium, do we need do the same for the alpine image? not sure if it will need other packages.